Identity Provider Info

quincydejong opened the pull request:
Users should be able to navigate to `/language/identity-provider-slug` and get some general info on what to do next.

quincydejong committed:
chore: init React and ASP.NET core Visual Studio project template

Created from Visual Studio (17.13.6) new project template

quincydejong committed:
chore: remove template code

quincydejong committed:
chore: import assets from IBAN issuer

CSS, fonts, images and button svg

quincydejong committed:
feat: add language based routing based on IBAN issuer

quincydejong committed:
feat: add IdentityProviderService

quincydejong committed:
feat: add IdentityProviderController

For the UI to access identity providers by slug using the backend API

quincydejong committed:
chore: setup backend testing project

With xUnit, NSubstitute and Shouldly

quincydejong committed:
test: add initial IdentityProviderService tests

quincydejong committed:
feat: proxy frontend `/api/*` calls to backend

quincydejong committed:
feat: setup initial frontend

Replacing the original forecast template code using a modified version of the IBAN issuer CSS

quincydejong committed:
build: setup docker

quincydejong committed:
docs: add initial README.md

quincydejong committed:
docs: add json setup, prerequisites and apply formatting to README.md

DibranMulder commented on GIAP.Server/Services/IdentityProviderService.cs:
```diff
@@ -0,0 +1,46 @@
+
+namespace GIAP.Server.Services;
+
+/**
```
Normally you would use `/// `for comments in C#

DibranMulder commented on giap.client/CHANGELOG.md:
```diff
@@ -0,0 +1,16 @@
+This file explains how Visual Studio created the project.
+
+The following tools were used to generate this project:
```
Changelogs can be generated with releases in Github automatically. Its combines all the PR's and then it generates a changelog.

DibranMulder commented on giap.client/index.html:
```diff
@@ -0,0 +1,13 @@
+    <meta charset="UTF-8" />
+    <link rel="icon" type="image/svg+xml" href="/vite.svg" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>Vite + React + TS</title>
```
Dont forget to change the title

DibranMulder commented on giap.client/src/App.tsx:
```diff
@@ -0,0 +1,52 @@
+    let {slug} = useParams();
+
+    useEffect(() => {
+        populateIdentityProviderData();
```
What happens when the API is not accessible? Error handling.

DibranMulder commented on giap.client/vite.config.ts:
```diff
@@ -0,0 +1,61 @@
+import child_process from 'child_process';
+import { env } from 'process';
+
+const baseFolder =
```
Rather use `.env` files with environment specific API endpoints. Like used here: https://github.com/privacybydesign/yivi-portal/blob/f1278ad7728602e060dd6ef8ba33834b206b1716/portal_spa/src/services/axiosInstance.ts#L8
Also https is not needed on localhost.

DibranMulder commented on giap.client/src/main.tsx:
```diff
@@ -0,0 +1,35 @@
+    <StrictMode>
+        <BrowserRouter>
+            <Routes>
+                <Route path=":lang/*" element={<LanguageRouter/>}/>
```
I think you are missing a redirect for a defualt language. What if someone navigates to the root of this website then the user should be rerouted to /en or /nl

quincydejong committed:
fix: update comment style

quincydejong committed:
chore: remove template CHANGELOG.md's

quincydejong committed:
fix: update comment style in test

quincydejong committed:
fix: replace template title

Added setting the title based on IdP to backlog

quincydejong committed:
chore: update GIAP to use http for local development

Also simplifies vite.config.ts by removing template https cert configuration

quincydejong committed:
fix: redirect /:slug to /:lang/:slug

Currently no redirect is applied when the user navigates to /:slug

quincydejong committed:
feat: add basic error handling if API is not accessible

DibranMulder commented on giap.client/vite.config.ts:
```diff
@@ -0,0 +1,23 @@
+    server: {
+        proxy: {
+            '^/api.*': { // regex to proxy any api call
+                target: 'http://localhost:7179',
```
This probably needs to be configurable

quincydejong replied:
I agree, added to the backlog for now to merge in a future PR.

DibranMulder commented on giap.client/src/i18n.tsx:
```diff
@@ -0,0 +1,37 @@
+            nl: {
+                translation: {
+                    login: "Log in om attributen te laden",
+                    info: "Attributen kunnen eenvoudig in de Yivi app geladen worden. Hiervoor dient u in te loggen in via de knop hieronder en eventueel toestemming te geven. We gebruiken deze gegevens alleen tijdens het laden van de attributen in de app.",
```
`hiervoor dient u in te loggen in` loopt niet.

DibranMulder commented on GIAP.Server/Services/IdentityProviderService.cs:
```diff
@@ -0,0 +1,44 @@
+
+namespace GIAP.Server.Services;
+
+/// Class is used as a singleton, see Program.cs
```
Comments moeten in dit formaat, zou automatisch moeten gaan in visual studio (code) https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/xmldoc/recommended-tags

quincydejong committed:
feat: add xml formatted comments

Based on recommendations here: https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/xmldoc/recommended-tags

quincydejong committed:
fix: typo
